src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -89,34 +29,111 @@ const propTypes = {
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // Email of the logged in person
+        email: PropTypes.string,
+    }),
 };
 
 const defaultProps = {
     reportAction: {},
     isMini: false,
     isVisible: false,
+    session: {},
 };
 
-const ReportActionContextMenu = (props) => {
-    const wrapperStyle = getReportActionContextMenuStyles(props.isMini);
-    return props.isVisible && (
-        <View style={wrapperStyle}>
-            {CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
-                <ReportActionContextMenuItem
-                    icon={contextAction.icon}
-                    text={contextAction.text}
-                    successIcon={contextAction.successIcon}
-                    successText={contextAction.successText}
-                    isMini={props.isMini}
-                    onPress={() => contextAction.onPress(props.reportAction)}
-                    key={contextAction.text}
-                />
-            ))}
-        </View>
-    );
-};
+function copyToClipboard(reportAction) {
+    // If return value is true, we switch the `text` and `icon` on
+    // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+    // the `text` and `icon`
+    const message = _.last(lodashGet(reportAction, 'message', null));
+    const html = lodashGet(message, 'html', '');
+    const text = lodashGet(message, 'text', '');
+    const isAttachment = _.has(reportAction, 'isAttachment')
+        ? reportAction.isAttachment
+        : isReportMessageAttachment(text);
+    if (!isAttachment) {
+        Clipboard.setString(text);
+    } else {
+        Clipboard.setString(html);
+    }
+}
+
+class ReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        /**
```

Comment:
This should be an `//` comment (I thought I commented about this already)

Reply:
yep makes sesnse, I should've replaced the comment type when I copied this over.

Reply:
If I move this back outside the class, does it make sense to leave the comment as you suggest here though

Reply:
I don't really feel like this comment is very valuable (because it explains "what", and it's pretty obvious already what it is) so I think it's fine to just remove it.